From 28efca603bf11f361eb741d23ef181a3b6905f74 Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Thu, 11 May 2023 13:51:25 -0400
Subject: [PATCH] intel: Disable shader cache when executing intel_clc during
 the build

With the shader cache enabled, intel_clc attempts to write to ~/.cache.
Many distributions' build systems limit file-system access, and will
kill the process thus causing the build to fail.

Fixes: 639665053fa ("anv/grl: Build OpenCL kernels")
---
 src/intel/vulkan/grl/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/intel/vulkan/grl/meson.build b/src/intel/vulkan/grl/meson.build
index 3d7567cfc13..c0056b3493d 100644
--- a/src/intel/vulkan/grl/meson.build
+++ b/src/intel/vulkan/grl/meson.build
@@ -142,6 +142,7 @@ foreach t : [['125', 'gfx125', 'dg2']]
                                    # without modifying grl source code, remove
                                    # if fixed there
       ],
+      env: ['MESA_SHADER_CACHE_DISABLE=true'],
       depends : [prog_intel_clc]
     )
   endforeach
-- 
2.39.3

